### PR TITLE
Move gamedataChanged test before this.state is overwritten

### PR DESCRIPTION
**Pull request by roy7**
Noticed this when trying to debug https://github.com/online-go/gtp2ogs/issues/256

Was this way since original PR https://github.com/online-go/gtp2ogs/pull/193

### CODE REVIEW

File: game.js [hunk 1]
```diff
@@ -41,7 +41,10 @@ class Game {
 
             //this.log("Gamedata:", JSON.stringify(gamedata, null, 4));
 
+            // Test for chanaged state before overwriting prior state with gamedata packet
```
Comment (wonderingabout) [inline game.js:44]: small typo there `changed` :sweat_smile:

Comment (roy7) [reply game.js]: Thanks.

**Comment by sanderland** [comment]
could this be a clearer check at the top like
```
if this.state
  if equal, return / ignore
  if not, restart bot
```
or is doing half the function somehow necessary?

**Comment by roy7** [comment]
It could be changed that way. This is just how it was originally added so noone ever changed it. What your change would so is mean we don't need to check if the bot is running if we already knew this.state was null (since a null state can't have a running bot, right?). 

We could perhaps move it around now, or in a separate PR. Since this PR needs updated tests, I'm leaning towards minimal changes here to update the functionality and we can reformat later?

**Comment by roy7** [comment]
I'm referring to https://github.com/online-go/gtp2ogs/pull/268 which make this PR obsolete.

**Pull request closed without merging**